### DISCUSSION TITLE
Rig connection to pihpsdr and linhpsdr using TCP in hamlib

### DISCUSSION THREAD

**Discussion by rsaue** [foldynl/QLog#231] [Ideas]
Hi,
It would ble nice to have QLog utilize hamlib option on cat control with OpenHPSDR rigs on linux (and mac) using TCP. It is enabled in later versions of hamlib. In qlog the Openhpsdr pihpsdr option is for serial port connection only. 

73 de LB2EG Richard

**Comment by foldynl** [comment]
Please, could you send me any link or specification where it is described that OpenHPSDR can be connected via HAMLib and TCP? Maybe I misunderstood, but this radio is marked as `SERIAL` in the current hamlib source code (`port_type`).

```    RIG_MODEL(RIG_MODEL_HPSDR),
    .model_name = "PiHPSDR",
    .mfg_name =  "OpenHPSDR",
    .version =  BACKEND_VER ".1",
    .copyright =  "LGPL",
    .status =  RIG_STATUS_STABLE,
    .rig_type =  RIG_TYPE_TRANSCEIVER,
    .ptt_type =  RIG_PTT_RIG,
    .dcd_type =  RIG_DCD_RIG,
    .port_type =  RIG_PORT_SERIAL,
```

**Reply by rsaue** [reply]
Hi,
I am using lastest hamlib from github, where the TCP connection for pihpsdr Openhpsdr pihpsdr works well. I am not sure which version of hamlib this option was included for the first time, but i guess approx. mid 2021.  I enclose a screenshot with text console that shows cqrlog handshake/connection with pihpsdr using TCP port 19090. 

73 de LB2EG Richard

![Skjermbilde fra 2023-07-30 20-18-52](https://github.com/foldynl/QLog/assets/22942440/7d277551-a5f8-487a-9157-1eb58b09de4b)

**Comment by foldynl** [comment]
I will try to contact Hamlib maintainers to find out what the best method is to find whether RIG is a serial or network RIG.

**Comment by foldynl** [comment]
From the reaction of people from Hamlib (Hamlib/Hamlib#1352), I deduce that Hamlib is designed for WSJTX - it is only a joke. However, I found out that it is not possible to determine the type of RIG from Hamlib, which is quite unusual for such a wide-used library, that no one has solved this problem.

I will try to designed something more user-friendly for QLog

**Comment by rsaue** [comment]
Hi Ladislav, 
Thanks for adding TCP CAT connection for pihpsdr OpenHPSDR in testing_0.27 branch. 

Plays nicely on my Ubuntu linux system with both linhpsdr and pihpsdr.  

I found your set up solution allowing the user to choose between network and serial setup quite nice. As the standard port setup in pihpsdr and linhpsdr is to use port 19090, this could also be the starting point in Qlog too (instead of port 4532)?

73 de LB2EG Richard

**Comment by foldynl** [comment]
Many thanks for your feedback.

I don't have a chance to try it, so it's good that you tried it. As for the port, unfortunately, the situation is quite complicated. I'll give you examples, maybe you'll have some ideas:

1) it is necessary to have some default value in QLog - the port must be filled
2) Each RIG has a different default port. It is not possible to set port 19090 as default when, for example, K4 has a default port of 9200
3) Hamlib does not provide anything RIG related information - if it is a Network/Serial attached RIG, the number of the default port, etc.

Based on the above, I decided to set 4532 as the default port used by NETRIG.
